src/foam/java/refinements.js

```diff
@@ -435,10 +435,10 @@ foam.CLASS({
       var m = info.getMethod('cast');
       m.body = 'return ( o instanceof Number ) ?'
             + '((Number)o).intValue() :'
-            + '(int)o;';
+            + '((Integer) o).intValue();';
```

Comment:
Integer, Byte, Short etc are already instances of Number are they not?  When would the second case ever be hit?